File: Onsite/Long/G.cpp
#include <bits/stdc++.h>
using namespace std;

using i64 = long long;

string an(int x, int n) {
  if (x == n) {
    return "sin(" + to_string(x) + ")";
  }

  return "sin(" + to_string(x) + ((x % 2) ? "-" : "+") + an(x + 1, n) + ")";
}

string sn(int x, int n) {

  if (n == 1)
    return "sin(1)+" + to_string(x);

  return "(" + sn(x + 1, n - 1) + ")" + an(1, n) + "+" + to_string(x);
}

void solve() {

  int n;
  cin >> n;

  string y = sn(1, n);

  cout << y << endl;
}

int main() {

  ios::sync_with_stdio(false);
  cin.tie(nullptr);

  solve();

  return 0;
}

File: Onsite/Long/g.py
def an(x: int, n: int):
    if x == n:
        return "sin(" + str(x) + ")"

    if x % 2:
        sign = "-"
    else:
        sign = "+"

    return "sin(" + str(x) + sign + an(x + 1, n) + ")"


def sn(x: int, n: int):
    if n == 1:
        return "sin(1)+" + str(x)
    return "(" + sn(x + 1, n - 1) + ")" + an(1, n) + "+" + str(x)


n: int = int(input())
y = sn(1, n)

print(y)
